lib/chef/azure/commands/enable.rb

```diff
@@ -117,27 +120,44 @@ def configure_chef_only_once
         f.write(@validation_key)
       end
 
-      # Write client.rb
-      File.open("#{bootstrap_directory}/client.rb", "w") do |f|
-        f.write(override_clientrb_file(@client_rb))
-      end
-      # write the first_boot.json
-      File.open("#{bootstrap_directory}/first-boot.json", "w") do |f|
-        f.write(<<-RUNLIST
-{
-"run_list": [#{ @run_list.empty? ? "" : escape_runlist(@run_list)}]
-}
-RUNLIST
-)
+      if windows?
+          # Write client.rb
+          File.open("#{bootstrap_directory}/client.rb", "w") do |f|
+            f.write(override_clientrb_file(@client_rb))
+          end
+          # write the first_boot.json
+          File.open("#{bootstrap_directory}/first-boot.json", "w") do |f|
+            f.write(<<-RUNLIST
+    {
+    "run_list": [#{ @run_list.empty? ? "" : escape_runlist(@run_list)}]
+    }
+    RUNLIST
+    )
+          end
       end
 
       # run chef-client for first time with no runlist to register the node
       puts "Running chef client for first time with no runlist..."
 
       begin
-        params = " -c #{bootstrap_directory}/client.rb -E _default -L #{@azure_plugin_log_location}/chef-client.log --once "
-        result = shell_out("chef-client #{params}")
-        result.error!
+        if windows?
+          params = " -c #{bootstrap_directory}/client.rb -E _default -L #{@azure_plugin_log_location}/chef-client.log --once "

```

Comment:
For windows we need windows bootstrap context and template from knife-windows. So we are implementing this for linux first.